### PR TITLE
Add SCM certificate management support

### PR DESCRIPTION
**Pull request by chengxiangdong**
**What this PR does / why we need it**:

Add SCM certificate management support

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: openstack/scm/v3/certificates/results.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	NotBefore            string             `json:"not_before"`
+	NotAfter             string             `json:"not_after"`
+	ValidityPeriod       int8               `json:"validity_period,omitempty"`
+	ValidationMethod     string             `json:"validation_method"`
+	DomainType           string             `json:"domain_type"`
+	Domain               string             `json:"domain"`
+	Sans                 string             `json:"sans"`
+	DomainCount          int8               `json:"domain_count,omitempty"`
+	WildcardCount        int8               `json:"wildcard_count,omitempty"`
+	AuthentificationList []Authentification `json:"authentification,omitempty"`
```
Comment (Lance52259) [inline openstack/scm/v3/certificates/results.go:44]: I think `Authentifications` is better.

File: openstack/scm/v3/certificates/requests.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+// PushOptsBuilder is the interface options structs have to satisfy in order
+// to be used in the Push operation in this package.
+type PushOptsBuilder interface {
+	ToCertificatePushMap() (map[string]interface{}, error)
+}
+
+// PushOpts is the struct be used in the Import operation
+type PushOpts struct {
+	TargetProject string `json:"target_project"`
```
Comment (Lance52259) [inline openstack/scm/v3/certificates/requests.go:54]: Does `TargetProject` has required label?

Comment (chengxiangdong) [reply openstack/scm/v3/certificates/requests.go:54]: when `TargetService` is Enhance_ELB or WAF,`TargetProject` is required.

File: openstack/scm/v3/certificates/results.go (hunk 1 above)
Comment (chengxiangdong) [reply openstack/scm/v3/certificates/results.go]: got it!

File: openstack/scm/v3/certificates/results.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	Domain            string             `json:"domain"`
+	Sans              string             `json:"sans"`
+	DomainCount       int8               `json:"domain_count,omitempty"`
+	WildcardCount     int8               `json:"wildcard_count,omitempty"`
+	Authentifications []Authentification `json:"authentification,omitempty"`
+}
+
+// CertificatePage is the page returned by a pager when traversing over a
+// collection of certificates.
+type CertificatePage struct {
```
Comment (ShiChangkuo) [inline openstack/scm/v3/certificates/results.go:49]: defined but not used

File: openstack/scm/v3/certificates/results.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	IssueTime         string             `json:"issue_time"`
+	NotBefore         string             `json:"not_before"`
+	NotAfter          string             `json:"not_after"`
+	ValidityPeriod    int8               `json:"validity_period,omitempty"`
+	ValidationMethod  string             `json:"validation_method"`
+	DomainType        string             `json:"domain_type"`
+	Domain            string             `json:"domain"`
+	Sans              string             `json:"sans"`
+	DomainCount       int8               `json:"domain_count,omitempty"`
+	WildcardCount     int8               `json:"wildcard_count,omitempty"`
```
Comment (ShiChangkuo) [inline openstack/scm/v3/certificates/results.go:43]: please change **int8** to **int**

File: openstack/scm/v3/certificates/results.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+	Domain      string `json:"domain"`
+}
+
+// The struct defines the information about the imported certificate.
+// This struct is used by the API for obtain certificate information and importing.
+type Certificate struct {
+	// the importing API only uses certificate_id.
+	CertificateId     string             `json:"certificate_id"`
+	Id                string             `json:"id"`
+	Name              string             `json:"name,omitempty" required:"true"`
```
Comment (ShiChangkuo) [inline openstack/scm/v3/certificates/results.go:23]: **required** and **omitempty** are unnecessary for a response result.

File: openstack/scm/v3/certificates/results.go (hunk 4 above)
Comment (chengxiangdong) [reply openstack/scm/v3/certificates/results.go:59]: fixed

File: openstack/scm/v3/certificates/results.go (hunk 3 above)
Comment (chengxiangdong) [reply openstack/scm/v3/certificates/results.go]: Has been removed

**Pull request merged**